node/src/service.rs

```diff
@@ -298,7 +457,15 @@ async fn start_node_impl(
         .map(|e| e.relay_chain.clone())
         .ok_or_else(|| "Could not find relay_chain extension in chain-spec.")?;
 
-    let params = new_partial(&parachain_config)?;
+    // Channel to send messages to start/stop container chains
+    let (cc_spawn_tx, cc_spawn_rx) = unbounded_channel();
+    // If this node was started without a `container_chain_config`, we don't support collation
+    // on container chains, so there is no need to detect changes to assignment
+    let params = if container_chain_config.is_some() {
```

Comment:
In this case, what happens if I start as a Tanssi-collator, then I get moved to a container-chain? what happens if I start without the container_chain_config, because I am a tanssi-collator, but then I get re-assigned?

Reply:
If you start without the `container_chain_config`, the collator will always try to collate on Tanssi.

It shouldn't be too hard to fix it (we just need to create a default config if it's missing, or even panic and ask the user to provide it), but I would prefer to leave it for a future PR.

For now if you want I can start all the collators with a `container_chain_config`, by modifying the args to `-- --tmp`.

Reply:
I see perfect. Yeah I agree we can leave it for future PR, let's also open an issue for this. For me it should be that:
- If parachain_config.is_authority() is true, then we should always spawn new_partial_orchestrator, because you might get re-assigned just because you are a collator